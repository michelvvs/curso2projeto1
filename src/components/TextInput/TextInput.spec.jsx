import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { TextInput } from '.';

describe('<TextInput />', () => {
  it('should have a value in searchValue', () => {
    const fn = jest.fn();
    //const {debug} =
    render(<TextInput handleChange={fn} searchValue="teste" />);
    //debug();
    const input = screen.getByPlaceholderText(/digite sua busca aqui/i);
    expect(input).toBeInTheDocument();
    expect(input.value).toBe('teste');
  });

  it('should call handleChange function on it key pressed', () => {
    const fn = jest.fn();
    //const {debug} =
    render(<TextInput handleChange={fn} />);
    const input = screen.getByPlaceholderText(/digite sua busca aqui/i);
    const typedValue = 'Typed Value';
    userEvent.type(input, typedValue);
    screen.debug(input);

    expect(input.value).toBe(typedValue);
    expect(fn).toBeCalledTimes(typedValue.length);
  });

  it('should match the snapshot', () => {
    const fn = jest.fn();
    //const {debug} =
    const { container } = render(<TextInput handleChange={fn} />);
    expect(container).toMatchSnapshot();
  });
});
