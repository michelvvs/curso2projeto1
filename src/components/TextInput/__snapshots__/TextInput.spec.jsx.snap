// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> should match the snapshot 1`] = `
<div>
  <input
    class="text-input"
    placeholder="digite sua busca aqui"
    type="search"
    value=""
  />
</div>
`;
